extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/retry/backoff/RetryOnClassBackoffOnMethodTest.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.smallrye.faulttolerance.test.retry.backoff;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import javax.enterprise.inject.spi.DefinitionException;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class RetryOnClassBackoffOnMethodTest {
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(RetryOnClassBackoffOnMethodService.class))
+            .assertException(e -> {
+                assertEquals(DefinitionException.class, e.getClass());
+                assertTrue(e.getMessage().contains("Backoff annotation"));
+                assertTrue(e.getMessage().contains("@Retry is missing"));
```

Comment:
Just out of curiosity, why not to allow the `@FibonacciBackoff` to be declared on the class and if so then apply to all methods annotated with `@Retry`?

Reply:
I find that a little hard to understand.

Ideally, I'd want something like `@Retry(backoff = @[Exponential|Fibonacci|Custom]Backoff(...))` or `@Retry(exponentialBackoff = null, fibonacciBackoff = @FibonacciBackoff(...))`, but nothing like that is possible.

So what I'm doing here is as close to that as possible -- the backoff annotation must be present on the same element as `@Retry`.

Loosening constraints is always possible.